src/identity/configuration/IdentityProviderFactory.ts

```diff
@@ -136,9 +139,44 @@ export class IdentityProviderFactory implements ProviderFactory {
     const provider = new Provider(this.baseUrl, config);
     provider.proxy = true;
 
+    this.alwaysRenderErrors(provider);
+
     return provider;
   }
 
+  /**
+   * In the `configureErrors` function below, we configure the `renderError` function of the provider configuration.
+   * This function is called by the OIDC provider library to render errors,
+   * but only does this if the accept header is HTML.
+   * Otherwise, it just returns the error object iself as a JSON object.
+   * See https://github.com/panva/node-oidc-provider/blob/0fcc112e0a95b3b2dae4eba6da812253277567c9/lib/shared/error_handler.js#L48-L52.
+   *
+   * In this function we override the `ctx.accepts` function
+   * to make the above code think HTML is always requested there.
+   * This way we have full control over error representation as configured in `configureErrors`.
+   * We still check the accept headers ourselves so there still is content negotiation on the output,
+   * the client will not simply always receive HTML.
+   *
+   * Should this part of the OIDC library code ever change, our function will break,
+   * at which point behaviour will simply revert to what it was before.
+   */
+  private alwaysRenderErrors(provider: Provider): void {
```

Comment:
```suggestion
  private captureErrorResponses(provider: Provider): void {
```

or, if we change to everything

```suggestion
  private captureResponses(provider: Provider): void {
```

Reply:
…alternatively, should this just be part of `configureErrors`? Then it's all easy to follow.

Reply:
> …alternatively, should this just be part of `configureErrors`? Then it's all easy to follow.

Problem is that we need to update the `Provider` object, which gets created using a configuration. `configureErrors` modifies that configuration object (to define the `renderError` function) so at that point we don't have the provider object yet.

Reply:
Clear; `captureErrorResponses` or `captureErrors` seems appropriate.